module/apmawssdkgo/session.go

```diff
@@ -79,8 +81,10 @@ func build(req *request.Request) {
 		return
 	}
 
+	if spanSubtype == serviceSNS && !supportedSNSMethod(req) {
+		return
+	}
```

Comment:
This is starting to get repetitive (see line below). I kept it as a separate `if` statement, but if you have any preferences on how to collapse this, I'm happy to change it.